### PR TITLE
Login signup component closes #31 closes #36

### LINKED ISSUE #31: User login, logout, and signup components
Add a user login pages as well as a separate user signup page.

### LINKED ISSUE #36: Add user update/information page
Add page where the user can see their profile details, as well as update their personal account information.

### PR DESCRIPTION
**Pull request by CB6670**
User can now signup, login, or logout.

### CODE REVIEW

File: client/components/SignUpForm.js [hunk 1]
```diff
@@ -1,11 +1,95 @@
...
+
+handleSubmit(ev){
+  ev.preventDefault();
+  const sObj = {}
+  for (let key in this.state){
+    if (key !== 'cPassword'){
+      sObj[key] = this.state[key]
+    }
+  }
+  axios.post('/api/users/signup', sObj)
```
Comment (o6nH) [inline client/components/SignUpForm.js]: Should this axios call also made an action that sets user the redux store's state?

File: server/db/models/User.js [hunk 2]
```diff
@@ -1,94 +1,120 @@
...
+}, {
+    hooks: {
+        beforeCreate: user =>  {
+            if (user.password){
+                user.password = hash(user.password);
+            } else {
+                return user
+            }
+        },
+        beforeUpdate: user => {
```
Comment (o6nH) [inline server/db/models/User.js]: Nice!

File: server/db/models/User.js [hunk 3]
```diff
@@ -1,94 +1,120 @@
...
+        email, 
+        password: hash(password)
+      }
+    })
+}
+
+User.signup = async function (user) {
+    //if guest and wants to sign up then update user
+    // if email is already taken the tell user that
+        //console.log('USER $$$$$$$$ ', user)
```
Comment (o6nH) [inline server/db/models/User.js]: Remove debugging logs after they're no longer needed

File: client/components/Navbar.js [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
 import {Link} from 'react-router-dom';
 import {connect} from 'react-redux';
+import Axios from 'axios';
+import {getCurrentUser} from '../actions';
 
 const marketName = 'ShopWare';
 // const marketLogo = './img/bottle_64.png';
 // const cartImg = './img/cart_64.png';
 
+const signOut = async (getUser) => {
```
[the remark this answers is not in the document]
Comment (CB6670) [reply client/components/Navbar.js]: With signOut, all I'm doing is deleting the session. and then replacing the user entirely in the userReducer with the getUser call. I don't think that it's being stored anywhere else on the front end.

File: client/components/SignUpForm.js (hunk 1 above)
Comment (CB6670) [reply client/components/SignUpForm.js]: I'm just setting the userId with the signUp call. Then after that, I call getUser and populate the store that way. I know it's not the most efficient way, but I felt getting a simple answer in place and getting the code out would be best.

**Pull request merged**